### ISSUE TITLE
Add UC Browser

### ISSUE THREAD

**Issue by laszbalo** [compat-table/compat-table#1001]
The crappiest browser I've ever tried, unfortunately very popular in some countries (India ~43%). What would it take to add it to the list? What can I do to make it happen?

With thanks,
Laszlo

**Comment by afmenez** [comment]
It is not that hard, you can download the sources, add the browser to environments.json, and add the results to each specific datafile.  Then rebuild the htmls and submit the patch.

**Comment by laszbalo** [comment]
@afmenez  Thank you. Sounds very straightforward. My first thing in the morning.

**Comment by n8schloss** [comment]
@laszbalo, to be clear when you mention UC browser you're talking about UC browser for Android? There's also UC Mini for Android and a desktop version of the browser and I don't think they are all running the same versions of the js engine.

**Comment by laszbalo** [comment]
@n8schloss Yes, I was talking about UC browser for Android. Luckily, their other browsers have not gained enough traction yet (hope they never will :smile:)
Not sure if UC mini can even run JS.

**Comment by n8schloss** [comment]
Cool, UC mini is definitely gaining traction so it might be worth adding it as well. I guess I'd just try to be clear which version of the browser this is in reference to when it get's added.

**Comment by laszbalo** [comment]
@n8schloss I hear you. Are you interested in the so called Speed mode, as well? Basically UC web uses proxies for pre-rendering the page and then sends down the data in binary format to the browser. Opera mini can do the same by enabling Extreme Mode.
I am writing a tiny app, that will add the required data to the data files as @afmenez suggested. As it is currently implemented, it can collect the necessary data from any connected user agent and update the data files for me. If I host it on the Internet, I think, it will work for both UC mini and Opera mini running in proxy mode.

**Comment by n8schloss** [comment]
@laszbalo, yeah I think it may make sense to breakup UC into three different categories then? UC Browser, UC Mini and UC Speed Mode? Also, that app sounds sweet!

**Comment by Jon-Galloway** [comment]
Curious as well.  Please advise.  8.75% of global market share is UC Browser according to http://gs.statcounter.com/ (right behind Chrome and Safari).

**Comment by alexander-akait** [comment]
@Jon-Galloway seems, we should do PR :smile:

**Comment by char8x** [comment]
@gclxry Given that the UC browser has a certain market share in the global market, will the UC team be willing to submit a PR to this repo? It is helpful for both Chinese and foreign developers. 😄

**Comment by ai** [comment]
Autoprefixer support all this browsers for CSS. And Can I Use sorry them too.

Right now because of missing them in Kangax Babel ignores this browsers and it is really bad.
